src/module.c

```diff
@@ -8475,23 +8475,38 @@ int moduleLoad(const char *path, void **module_argv, int module_argc) {
     }
     onload = (int (*)(void *, void **, int))(unsigned long) dlsym(handle,"RedisModule_OnLoad");
     if (onload == NULL) {
-        dlclose(handle);
         serverLog(LL_WARNING,
             "Module %s does not export RedisModule_OnLoad() "
             "symbol. Module not loaded.",path);
-        return C_ERR;
+        goto error;
     }
-    if (onload((void*)&ctx,module_argv,module_argc) == REDISMODULE_ERR) {
-        if (ctx.module) {
-            moduleUnregisterCommands(ctx.module);
-            moduleUnregisterSharedAPI(ctx.module);
-            moduleUnregisterUsedAPI(ctx.module);
-            moduleFreeModuleStructure(ctx.module);
+
+    if (onload((void*)&ctx,module_argv,module_argc) == REDISMODULE_ERR) goto error;
+
+    /* One of rdb_load and rdb_save can't be NULL and the other is not,
+     * same with aux_load and aux_save */
+    if (listLength(ctx.module->types)) {
```

Comment:
Actually， At beginning. I add check in RM_CreateDataType. But It can just return NULL to notify module developer.
```
    datatype = RedisModule_CreateDataType(ctx, "test___dt", 1, &datatype_methods);
    if (datatype == NULL)
        return REDISMODULE_ERR;
```
If module developer didn't check `datatype == NULL`. Module can be loaded with one method is defined and the other is not..

So I think add check in module loading.   What you think about which way is better